chalice/chalicelib/matrix_handler.py

```diff
@@ -0,0 +1,165 @@
+import os
+import tempfile
+import time
+import traceback
+import loompy
+
+from abc import ABC, abstractmethod
+from subprocess import call
+from typing import List
+from chalicelib import get_mtx_paths
+from chalicelib.config import MERGED_MTX_BUCKET_NAME, TEMP_DIR, hca_client, \
+    s3_blob_store, logger
+from chalicelib.request_handler import RequestHandler, RequestStatus
+
+
+class MatrixHandler(ABC):
+    """
+    A generic matrix handler for matrices concatenation
+    """
+
+    def __init__(self, suffix) -> None:
+        self._suffix = suffix
+
+    def _download_mtx(self, bundle_uuids: List[str], temp_dir: str) -> List[str]:
+        """
+        Filter for the matrix files within bundles, and download them locally.
+        :param bundle_uuids: A list of bundle uuids.
+        :param temp_dir: A temporary directory for storing all downloaded matrix files.
+        :return: A list of downloaded matrix files paths.
+        """
+        # Filter and download only matrices file that satisfies a specific suffix within bundles
+        for bundle_uuid in bundle_uuids:
+            dest_name = os.path.join(temp_dir, bundle_uuid)
+
+            try:
+                logger.info("Downloading matrices from bundle, {}, into {}."
+                            .format(bundle_uuid, dest_name))
+
+                hca_client.download(
+                    bundle_uuid=bundle_uuid,
+                    replica="aws",
+                    dest_name=dest_name,
+                    metadata_files=(),
+                    data_files=("*{}".format(self._suffix),)
+                )
+            except Exception as e:
```

Comment:
??

you can just remove the try-block then.

Reply:
Marcus also points that out. I will remove it.